### PR TITLE
React date/datetime field

### PR DESCRIPTION
**Pull request by phalkunz**
Issue: https://github.com/silverstripe/silverstripe-framework/issues/6140

### CODE REVIEW

File: client/src/components/DateField/DateField.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        localDate = dateObject.format(localFormat);
+      }
+    }
+    return localDate;
+  }
+
+}
+
+DateField.propTypes = {
+  lang: React.PropTypes.string,
```
Comment (tractorcow) [inline client/src/components/DateField/DateField.js]: Can you expand on this, and declare the `data` key (see my review on framework for context) with the necessary subkeys? We should also add top level standard keys here.

I'll create an issue for this.

Comment (tractorcow) [inline client/src/components/DateField/DateField.js]: See https://github.com/silverstripe/silverstripe-admin/issues/43

File: client/src/components/DateTimeField/DateTimeField.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+class DateTimeField extends DateField {
+
+  getInputProps() {
+    const placeholder = i18n.inject(
+      i18n._t('DateTimeField.DateTimeFormatExample', 'Example: {datetime}'),
+      { datetime: moment().endOf('month').format(localFormat) }
+    );
+    const props = {};
+    Object.assign(props, super.getInputProps());
```
Comment (tractorcow) [inline client/src/components/DateTimeField/DateTimeField.js]: Can we chain Object.assign()?

```
return Object.assign(
  {},
  super.getInputProps(),
  {
    type: this.props.html5 ? 'datetime-local' : 'text',
    placeholder,
  }
);
```

File: client/src/components/DateField/DateField.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  getInputProps() {
+    const placeholder = i18n.inject(
+      i18n._t('DateField.DateFormatExample', 'Example: {datetime}'),
+      { datetime: moment().endOf('month').format(localFormat) }
+    );
+    const props = {};
+
+    let val = this.props.value;
+
+    if (!this.props.html5 || (this.hasNativeSupport() && this.props.html5)) {
```
Comment (tractorcow) [inline client/src/components/DateField/DateField.js]: `this.props.data.html5` ?

Comment (chillu) [inline client/src/components/DateField/DateField.js]: Done

File: client/src/components/DateTimeField/DateTimeField.js (hunk 2 above)
Comment (chillu) [inline client/src/components/DateTimeField/DateTimeField.js]: Yep, I've done that now

**Pull request merged**